ui/gfxr_capture_worker.cpp

```diff
@@ -0,0 +1,247 @@
+    {
+        std::string get_timestamp_command = absl::StrCat("shell stat -c %Y ",
+                                                         m_source_capture_dir,
+                                                         "/",
+                                                         file_name.data());
+
+        absl::StatusOr<std::string> current_timestamp = device->Adb().RunAndGetResult(
+        get_timestamp_command);
+        if (!current_timestamp.ok())
+        {
+            qDebug() << "Failed to get timestamp for " << file_name.data() << ": "
+                     << current_timestamp.status().message().data();
+            return false;
+        }
+
+        if (current_timestamp->data() != timestamp)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+//--------------------------------------------------------------------------------------------------
+absl::StatusOr<int64_t> GfxrCaptureWorker::getGfxrCaptureDirectorySize(Dive::AndroidDevice *device)
+{
+    // Retrieve the names of the files in the capture directory on the device.
+    std::string                 ls_command = "shell ls " + m_source_capture_dir;
+    absl::StatusOr<std::string> ls_output = device->Adb().RunAndGetResult(ls_command);
+
+    if (!ls_output.ok())
+    {
+        std::cout << "Error getting capture files: " << ls_output.status().message() << std::endl;
+        return ls_output.status();
+    }
+
+    m_file_list = absl::StrSplit(std::string(ls_output->data()), '\n');
+
+    for (std::string &file_with_trailing : m_file_list)
+    {
+        // Windows-style line endings use \r\n. When absl::StrSplit splits by \n, the \r remains
+        // at the end of each line if the input string originated from a Windows-style line
+        // ending.
+        if (!file_with_trailing.empty() && file_with_trailing.back() == '\r')
+        {
+            file_with_trailing.pop_back();
+        }
+    }
+
+    while (true)
+    {
+        // Ensure that the .gfxa, .gfxr, and .png file sizes are set and neither is being written
+        // to.
+        int64_t                            size = 0;
+        std::map<std::string, std::string> current_timestamps;
+
+        for (std::string file : m_file_list)
+        {
+            std::string path = absl::StrCat(m_source_capture_dir, "/", file.data());
+
+            // Get the size of the file.
+            std::string get_file_size_command = "shell stat -c %s " + path;
+
+            // Get the timestamp for last time the file was updated.
+            std::string get_file_update_timestamp_command = "shell stat -c %Y " + path;
+
+            absl::StatusOr<std::string> str_num = device->Adb().RunAndGetResult(
+            get_file_size_command);
+            absl::StatusOr<std::string> file_update_timestamp = device->Adb().RunAndGetResult(
+            get_file_update_timestamp_command);
+            int64_t num = std::stoll(str_num->c_str());
```

Comment:
maybe better use absl::SimpleAtoi to avoid handling exception.

Reply:
Added